src/CodeGeneration.Roslyn.Engine/CompilationGenerator.cs

```diff
@@ -150,7 +150,7 @@ public async Task GenerateAsync(IProgress<Diagnostic> progress = null, Cancellat
                             catch (IOException ex) when (ex.HResult == ProcessCannotAccessFileHR && retriesLeft > 0)
                             {
                                 retriesLeft--;
-                                Task.Delay(200).Wait();
+                                await Task.Delay(200, cancellationToken);
                             }
                             catch (Exception ex)
```

Comment:
Should we maybe handle [`OperationCancelledException`](https://docs.microsoft.com/en-us/dotnet/api/system.operationcanceledexception?view=netframework-4.8) here as a special case then?

If the operation got cancelled, I'd not expect generation to continue on all the other files in queue.

Reply:
Proposal A: in case of a cancellation the `while` loop could be exited, then in the `foreach` the flow would be cancelled. Additionally after the `foreach` an `cancellationToken.ThrowIfCancellationRequested()` should be added.

Proposal B: add an filter `catch(Exception ex) when(!(ex is OperationCanceledException))`

After writing these proposals I prefer proposal B

Reply:
I think that's the simplest and most clear solution. And additionally, doesn't interfere with stacktraces, so an additional bonus.